lib/components/chip.ts

```diff
@@ -31,13 +33,11 @@ export interface ChipProps extends CommonComponentProps {
   noSchematicRepresentation?: boolean
   internallyConnectedPins?: string[][]
   externallyConnectedPins?: string[][]
-  connections?: Connections
+  connections?: Connections<PinLabel>
 }
 
-export type PinLabels = Record<
-  number | string,
-  string | readonly string[] | string[]
->
+export type PinLabels<PinLabel extends string | number = string | number> =
```

Comment:
imo this makes it a bit less confusing
```suggestion
export type PinLabelsProp<PinLabel extends string | number = string | number> =
```